code-challenges.js

```diff
@@ -2,35 +2,67 @@
 
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
-
-
-
+const secretWordGenerator = (string) => {
+	let newString = ""
+	for(let i = 0; i < string.length; i++){
+		if(string[i] == "a") newString += ("4")
+		else if(string[i] == "e") newString +=("3")
+		else if(string[i] == "i") newString += ("1")
+		else if(string[i] == "o") newString += ("0")
+		else newString += (string[i])
+	}
+	return newString
+}
 
 var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+console.log(secretWordGenerator(secretCodeWord1))
+console.log(secretWordGenerator(secretCodeWord2))
 // --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
 
-
-
-
+const wordsWithA = (arr) => {
+	let newArr = []
+	for(let i = 0; i < arr.length; i++){
+		if(arr[i].includes("A") || arr[i].includes("a")) newArr.push(arr[i])
+	}
+  return newArr
+}
```

Comment:
great! you can even say `if(arr[i].toLowerCase().includes("a")) newArr.push(arr[i])` to make it slightly shorter if you'd like :)